Framework/CurveFitting/test/Functions/AsymmetricPearsonVIITest.h

```diff
@@ -0,0 +1,596 @@
+  static AsymmetricPearsonVIITest *createSuite() { return new AsymmetricPearsonVIITest(); }
+  static void destroySuite(AsymmetricPearsonVIITest *suite) { delete suite; }
+
+  AsymmetricPearsonVIITest() : m_xValues(), m_10xValues() {
+    for (size_t i = 0; i < 2000; ++i) {
+      double x_i = -1 + 0.001 * static_cast<double>(i);
+      m_xValues.emplace_back(x_i);
+    }
+
+    for (size_t i = 0; i < 10; ++i) {
+      double x_i = -1 + 0.2 * static_cast<double>(i);
+      m_10xValues.emplace_back(x_i);
+    }
+  }
+
+  void testCategory() {
+    AsymmetricPearsonVII ap7;
+    TS_ASSERT_EQUALS(ap7.category(), "XrayDiffraction");
+  }
+
+  void testSetParameters() {
+    AsymmetricPearsonVII ap7;
+    const double ph = 0.9;
+    const double pc = 0.5;
+    const double width = 1.1;
+    const double ml = 2.0;
+    const double mr = 0.0;
+    ap7.initialize();
+    ap7.setHeight(ph);
+    ap7.setCentre(pc);
+    ap7.setFwhm(width);
+    ap7.setLeftShape(ml);
+    ap7.setRightShape(mr);
+
+    TS_ASSERT_DELTA(ap7.height(), ph, 1.0e-6);
+    TS_ASSERT_DELTA(ap7.centre(), pc, 1.0e-6);
+    TS_ASSERT_DELTA(ap7.fwhm(), width, 1.0e-6);
+    TS_ASSERT_DELTA(ap7.leftShape(), ml, 1.0e-6);
+    TS_ASSERT_DELTA(ap7.rightShape(), mr, 1.0e-6);
+  }
+
+  /** Test expected output
+   * @brief testExpectedOutput: test if the asymmetric Pearson VII
+   * function returns the expected values on the interval [-1, 1)
+   * and step 0.2
+   */
+  void testExpectedOutput() {
+    const double ph = 10.0;
+    const double pc = 0.0;
+    const double width = 0.1;
+    const double ml = 10.0;
+    const double mr = 1.0;
+
+    IPeakFunction_sptr ap7 = getInitializedAPVII(ph, pc, width, ml, mr);
+    FunctionDomain1DVector domain(m_10xValues);
+    FunctionValues values_ap7(domain);
+    ap7->function(domain, values_ap7);
+
+    double expected_output[10] = {1.86665e-14, 1.34216e-12, 2.85522e-10, 3.33551e-7, 0.00477417,
+                                  10.,         0.588235,    0.153846,    0.0689655,  0.0389105};
+
+    for (size_t i = 0; i < values_ap7.size(); ++i) {
+      TS_ASSERT_DELTA(expected_output[i], values_ap7[i], 1e-6);
+    }
+  }
+
+  /** Test integral
+   * @brief testIntegral: test if the integral of the asymmetric Pearson VII
+   * function over the interval (-100, 100) returns the expected value
+   */
+  void testIntegral() {
+    const double ph = 100.0;
+    const double pc = 0.0;
+    const double width = 0.7;
+    const double ml = 1.7;
+    const double mr = 2.0;
+    const double ll = -100.0;
+    const double ul = 100.0;
+    const double expected_value = 86.8874;
+
+    double num_integral = numerical_integrate_ap7(ph, pc, width, ml, mr, ll, ul);
+    TS_ASSERT_DELTA(num_integral, expected_value, 1.0e-4);
+  }
+
+  /** Test asymptotic behavior
+   * @brief testAsymptoticBehavior: test if the asymmetric Pearson VII
+   * behaves as Lorentzian at m = 1 and as Gaussian at m -> \infty
+   */
+  void testAsymptoticBehavior() {
```

Comment:
lets split the two limits into 2 tests. Then it is easier to identify what could have done wrong if there is a problem in the future.